packages/blockchain-wallet-v4-frontend/src/data/middleware/webSocket/activities/sagas.js

```diff
@@ -0,0 +1,59 @@
+import { prop } from 'ramda'
+import { call, put, select } from 'redux-saga/effects'
+
+import { sha256 } from '@core/walletCrypto'
+import { actions, selectors } from 'data'
+
+export default ({ socket }) => {
+  const authenticateSocket = function* () {
+    const guid = yield select(selectors.core.wallet.getGuid)
+    const sharedKey = yield select(selectors.core.wallet.getSharedKey)
+
+    const guidHash = sha256(guid).toString('hex')
+    const sharedKeyHash = sha256(sharedKey).toString('hex')
+
+    const authParams = {
+      action: 'subscribe',
+      auth: {
+        guidHash,
+        sharedKeyHash
+      },
+      channel: 'activity',
+      params: {
+        acceptLanguage: 'en-GB;q=1.0, en',
+        fiatCurrency: 'GBP',
+        timezoneIana: 'Europe/London'
+      }
+    }
+
+    socket.send(JSON.stringify(authParams))
+  }
+
+  const onOpen = function* () {
+    yield call(authenticateSocket)
+  }
+
+  const onAuth = function* () {}
```

Comment:
Do you need these methods if they don't do anything?

Reply:
I'll be addressing further WS implementations on the upcoming [PR](https://github.com/blockchain/blockchain-wallet-v4-frontend/pull/5653)